src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/SdkDirectoryWorkloadManifestProvider.cs

```diff
@@ -150,7 +175,7 @@ void ProbeDirectory(string manifestDirectory, string featureBand)
                 (string? id, string? finalManifestDirectory) = ResolveManifestDirectory(manifestDirectory);
                 if (id != null && finalManifestDirectory != null)
                 {
-                    AddManifest(id, finalManifestDirectory, featureBand);
+                    AddManifest(id, finalManifestDirectory, featureBand, Path.GetFileName(manifestDirectory));
```

Comment:
If the workload manifest is not side-by-side, then this won't pass the correct manifest version.  In that case you would need to read the actual manifest file to get the version.